src/commands.c

```diff
@@ -86,6 +86,12 @@ static void cmd_start(Tox *m, int friendnum, int argc, char (*argv)[MAX_COMMAND_
     }
 
     tox_friend_send_message(m, friendnum, TOX_MESSAGE_TYPE_NORMAL, (uint8_t *) outmsg, strlen(outmsg), NULL);
+
+    char *play_msg = "CSBot Has started, It's time to PLAY!";
+    for(int i = Tox_Bot.num_friends; i >= 0; i-- ) {
```

Comment:
Because toxcore doesn't do any message queuing, so this will still send to
just online friends nearly, if not more efficiently.

My other thought was to track online/offline friends in a separate list.
But the only place to do that would be inside the friend status callback.
And that gets called more much more often then the server starting/stopping.
On Apr 30, 2016 9:39 PM, "JFreegman" <notifications@github.com> wrote:

> In src/commands.c
> <https://github.com/zodiaclabs/CSbot/pull/1#discussion_r61679425>:
>
> > @@ -86,6 +86,12 @@ static void cmd_start(Tox *m, int friendnum, int argc, char (*argv)[MAX_COMMAND_
> >      }
> >
> >      tox_friend_send_message(m, friendnum, TOX_MESSAGE_TYPE_NORMAL, (uint8_t *) outmsg, strlen(outmsg), NULL);
> > +
> > +    char *play_msg = "CSBot Has started, It's time to PLAY!";
> > +    for(int i = Tox_Bot.num_friends; i >= 0; i-- ) {
>
> Why do you send this to all friends instead of just online friends?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zodiaclabs/CSbot/pull/1/files/c29d694156fe5d8df6160c5beb7421136530f47e#r61679425>
>